### PR TITLE
[openssl] Fix installation of openssl[core,tools]

### LINKED ISSUE #31170: [openssl] 3.1.0#3 on Windows doesn't build with "tools" but without "fips"
Package: openssl[core,tools]:x64-windows -> 3.1.0#3

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.35.32215.0
-    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: e52999ee1 2023-04-27 (8 hours ago)

**To Reproduce**

`vcpkg install openssl[core,tools]`

**Failure logs**

```
-- Using cached openssl-openssl-openssl-3.1.0.tar.gz.
-- Cleaning sources at C:/buildbot/msbuild/openvpn-build/src/vcpkg/buildtrees/openssl/src/nssl-3.1.0-1ebd9e680e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/buildbot/msbuild/openvpn-build/src/vcpkg/downloads/openssl-openssl-openssl-3.1.0.tar.gz
-- Applying patch disable-apps.patch
-- Applying patch disable-install-docs.patch
-- Applying patch script-prefix.patch
-- Applying patch windows/install-layout.patch
-- Applying patch windows/install-pdbs.patch
-- Applying patch unix/android-cc.patch
-- Applying patch unix/move-openssldir.patch
-- Applying patch unix/no-empty-dirs.patch
-- Applying patch unix/no-static-libs-for-shared.patch
-- Using source at C:/buildbot/msbuild/openvpn-build/src/vcpkg/buildtrees/openssl/src/nssl-3.1.0-1ebd9e680e.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows
-- Getting CMake variables for x64-windows
-- Prerunning x64-windows-dbg
-- Building x64-windows-dbg
-- Prerunning x64-windows-rel
-- Building x64-windows-rel
CMake Error at ports/openssl/windows/portfile.cmake:92 (file):
  file RENAME failed to rename

    C:/buildbot/msbuild/openvpn-build/src/vcpkg/packages/openssl_x64-windows/fipsmodule.cnf

  to

    C:/buildbot/msbuild/openvpn-build/src/vcpkg/packages/openssl_x64-windows/tools/openssl/fipsmodule.cnf

  because: The system cannot find the file specified.



Call Stack (most recent call first):
  ports/openssl/portfile.cmake:68 (include)
  scripts/ports.cmake:147 (include)



```

**Additional context**

### PR DESCRIPTION
**Pull request by flichtenheld**
Do not try to install file that does not exist.

Fixes #31170

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/openssl/windows/portfile.cmake [hunk 1]
```diff
@@ -89,7 +89,9 @@ set(scripts "bin/c_rehash.pl" "misc/CA.pl" "misc/tsget.pl")
 if("tools" IN_LIST FEATURES)
     file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
     file(RENAME "${CURRENT_PACKAGES_DIR}/openssl.cnf" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/openssl.cnf")
-    file(RENAME "${CURRENT_PACKAGES_DIR}/fipsmodule.cnf" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/fipsmodule.cnf")
+    if("fips" IN_LIST FEATURES)
+	file(RENAME "${CURRENT_PACKAGES_DIR}/fipsmodule.cnf" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/fipsmodule.cnf")
+    endif()
```
Comment (dg0yt) [inline ports/openssl/windows/portfile.cmake:94]: You are creating the next bug...
The fips-related lines must be moved out of the tools-related block.

Comment (flichtenheld) [reply ports/openssl/windows/portfile.cmake:94]: Why? Shouldn't this command be limited to when both tools and fips are enabled? The file is explicitly deleted when tools is not enabled.

Comment (flichtenheld) [reply ports/openssl/windows/portfile.cmake:94]: With my patch the file is installed into tools iff both tools and fips are enabled. It is deleted when fips is enabled but tools not, And it is never generated if fips is not enabled. Which of these cases do you want to have handled differently?

Comment (dg0yt) [reply ports/openssl/windows/portfile.cmake:94]: Why should it be deleted at all for `openssl[fips]`?

Comment (dg0yt) [reply ports/openssl/windows/portfile.cmake:94]: > Why should it be deleted at all for `openssl[fips]`?

I see, it is about to be copied to the tools directory.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**